src/Hazelcast.Net.Tests/Core/ClientVersionTests.cs

```diff
@@ -50,15 +51,32 @@ public void ClientVersionReturnsVersion()
 
             // both present, ignores version
             Assert.That(ClientVersion.GetVersion(new AssemblyInformationalVersionAttribute("1.2.3"), new AssemblyVersionAttribute("4.5.6")), Is.EqualTo("1.2.3"));
+
+
+            Assert.That(Authenticator.ClientVersion, Is.EqualTo(ClientVersion.MajorMinorPatchVersion));
+            var assemblyVersion = typeof(ClientVersion).Assembly.GetName().Version;
+            Assert.That(Authenticator.ClientVersion, Is.EqualTo(assemblyVersion.Major + "." + assemblyVersion.Minor + "." + assemblyVersion.MajorRevision));
+            Assert.That(Authenticator.ClientVersion.Split('.').Length, Is.GreaterThanOrEqualTo(3));
+        }
+
+
+        [TestCase("1.2.3", "1.2.3")]
+        [TestCase("1.2.3-preview.0", "1.2.3")]
```

Comment:
You commented below java reports as 6.0.0-SNAPSHOT. In this case, I was expecting to see this test expects to be `1.2.3-preview.0`

Can you also add the `6.0.0-SNAPSHOT` test case?

Reply:
This is only for `ClientVersion.GetMajorMinorPatchVersion`. I've introduced another method `ClientVersion.GetSemVerWithoutBuildingMetadata` which has tests below. Also, added the case.